openedx/features/survey_report/context_processors.py

```diff
@@ -20,15 +20,19 @@ def admin_extra_context(request):
     The current treshhold to show the banner is one month but this can be redefined in the future
 
     """
+    show_survey_report_banner = False
```

Comment:
you could do the `settings.ENABLE_SURVEY_REPORT` read here and if it is false, return inmediately. This way if the flag if off then the query is also not running.

Reply:
Agreed. It also simplifies the logic.

Reply:
I refactored the conditions to amke it more readable